src/contexts/Toast/ToastProvider.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { createContext, useContext, useEffect, useState, useRef } from 'react'
+import { Message, Transition } from 'semantic-ui-react'
+import { ToastProps, MessageProps, Position, bottomLeft } from './types'
+import { getStyleProps } from './helpers'
+import { nanoid } from 'nanoid'
+
+const TRANSITION_DURATION = 350 // ms
+
+const setTimeout = window.setTimeout // To ensure return type is number
+interface ToastProviderValue {
+  showToast: (...state: Partial<ToastProps>[]) => void
+  updateDefaults: (state: Partial<ToastProps>) => void
+}
+
+const ToastProviderContext = createContext<ToastProviderValue>({
+  showToast: () => {},
+  updateDefaults: () => {},
+})
+
+export function ToastProvider({ children }: { children: React.ReactNode }) {
+  const [toastDefaults, setToastDefaults] = useState<ToastProps>({
+    title: '',
+    text: '',
+    style: 'basic',
+    timeout: 5000,
+    clickable: true,
+    showCloseIcon: true,
+    position: bottomLeft,
+    uid: '',
+  })
+  const [toasts, setToasts] = useState<ToastProps[]>([])
+
+  const updateDefaults = (newDefaults: Partial<ToastProps>) => {
+    setToastDefaults((prevDefaults) => ({ ...prevDefaults, ...newDefaults }))
+  }
+
+  const showToast = (newToast: Partial<ToastProps> = {}) => {
+    setToasts((prevState) => [...prevState, { ...toastDefaults, ...newToast, uid: nanoid(8) }])
+  }
+
+  const removeToast = (uid: string) => {
+    setToasts((prevState) => prevState.filter((toast) => toast.uid !== uid))
+  }
+
+  return (
+    <ToastProviderContext.Provider
+      value={{
+        showToast,
+        updateDefaults,
+      }}
+    >
+      <div id="toast-container">
+        {Object.entries(Position).map(([key, positionClass]) => (
+          <div className={`block ${positionClass}`} key={key}>
+            {toasts
+              .filter((toast) => toast.position === Position[key as keyof typeof Position])
+              .map((toast) => (
+                <Toast toast={toast} removeToast={() => removeToast(toast.uid)} key={toast.uid} />
+              ))}
+          </div>
+        ))}
+      </div>
+      {children}
+    </ToastProviderContext.Provider>
```

Comment:
`toast-container` is the fixed-position full-size container
`block ${positionClass}` are the six blocks, one of each position (top-left, etc.), positioned absolutely relative to the toast container
`<Toast />` -- the individual toasts, positioned relative to their respective blocks
`{children}` -- the rest of the app